### ISSUE TITLE
empty reponses (received as an HttpEntity.Default), cannot be consumed without size limit

### ISSUE THREAD

**Issue by hochgi** [akka/akka#20208]
This scenario produces: ``` java.lang.IllegalArgumentException: requirement failed: contentLength must be positive (use `HttpEntity.empty(contentType)` for empty entities)``` in client side.

here's a failing test to prove the issue (will add to `akka.http.scaladsl.ClientServerSpec`):

```scala
"receive an empty response as unlimited size entity" in Utils.assertAllStagesStopped {
  new TestSetup {
    val (clientOut, clientIn) = openNewClientConnection()
    val (serverIn, serverOut) = acceptConnection()

    val clientOutSub = clientOut.expectSubscription()
    clientOutSub.sendNext(HttpRequest(uri = "/empty"))
    clientOutSub.sendComplete()

    val serverInSub = serverIn.expectSubscription()
    serverInSub.request(1)
    serverIn.expectNext().uri shouldEqual Uri(s"http://$hostname:$port/empty")

    val serverOutSub = serverOut.expectSubscription()
    serverOutSub.expectRequest()
    serverOutSub.sendNext(HttpResponse(StatusCodes.OK, List(RawHeader("X-Res-Type", "EMPTY")), HttpEntity.Default(ContentTypes.`text/plain(UTF-8)`, 0L, Source.empty[ByteString])))

    val clientInSub = clientIn.expectSubscription()
    clientInSub.request(1)
    val response = clientIn.expectNext()
    val unlimitedEmptyFut = response.entity.withSizeLimit(-1).dataBytes.runFold(ByteString(""))(_ ++ _)
    Await.result(unlimitedEmptyFut, 100.millis).utf8String shouldEqual ""

    serverIn.expectComplete()
    serverOutSub.expectCancellation()
    clientIn.expectComplete()

    connSourceSub.cancel()
  }
}
```

**Comment by drewhk** [comment]
Thanks for reporting!

**Comment by ktoso** [comment]
The test does not really make sense though... it never reaches the point the issue is about, please refer to my comment in: https://github.com/akka/akka/pull/20209#issuecomment-205789484 (let's discuss there), so we can fix or just close this issue.

**Comment by ktoso** [comment]
Seems to be resolved since 2.4.3
